Can -pull work with non-public images?

I have an image that is not publicly accessible.  I can pull it via `gcloud docker -- pull blah`, which is probably translated to `docker login` under the covers.  Is there a way to make this work with the structure test `-pull` flag?

If so, maybe document how do that.  Maybe I need to fiddle with permissions or environment variables.

If not, maybe document that, because the error message is really vague (`Step #0:     	docker_driver.go:241: Error creating container: no such image`) and doesn't log the actual image name it's trying.

Luckily, the workaround is fairly easy -- for my case, where I'm invoking from a cloudbuild.yaml, I add a cloudbuild step before the structure test step, that explicitly pulls the image and runs `/bin/true` in it.

Repro:

```shell
$ gcloud container builds submit --config=cloudbuild_test.yaml --substitutions=_DOCKER_NAMESPACE=gcr.io/cloud-python-runtime-qa,_TAG=2018-03-08-174800-ubuntu16 .
<snip>
Step #0: e116a9187528: Pull complete
Step #0: Digest: sha256:3e6162829f0adafeb764883a715628c91c09a5a01c5bb5a25f1662ac76bb173b
Step #0: Status: Downloaded newer image for gcr.io/gcp-runtimes/container-structure-test:v0.2.1
Step #0: Error pulling remote image gcr.io/cloud-python-runtime-qa/python:2018-03-08-174800-ubuntu16: API error (404): {"message":"pull access denied for gcr.io/cloud-python-runtime-qa/python, repository does not exist or may require 'docker login'"}
```

cloudbuild_test.yaml:
```
steps:
- name: gcr.io/gcp-runtimes/container-structure-test:v0.2.1
  args: [
   '-pull',
    '-test.v',
    '-image', 'gcr.io/cloud-python-runtime-qa/python:2018-03-08-174800-ubuntu16',
    '/workspace/tests/virtualenv/virtualenv_default.yaml',
    ]
```

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,5 +1,6 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "go_prefix")
 load("@io_bazel_rules_docker//container:container.bzl", "container_image")
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
 
 licenses(["notice"])
 
@@ -12,6 +13,20 @@ gazelle(
     build_tags = ["go1.7"],
     external = "vendored",
     prefix = "github.com/GoogleCloudPlatform/container-structure-test",
+)
+
+pkg_tar(
+    name = "cred_helper",
+    srcs = ["@docker_credential_gcr//:docker-credential-gcr"],
+    mode = "0555",
+    package_dir = "/usr/local/bin",
+)
+
+pkg_tar(
+    name = "default_creds",
+    srcs = [":config.json"],
+    mode = "0600",
+    package_dir = "/root/.docker",
 )
 
 sh_binary(
@@ -32,13 +47,21 @@ go_binary(
         "//vendor/github.com/sirupsen/logrus:go_default_library",
         "//vendor/gopkg.in/yaml.v2:go_default_library",
     ],
+    data = [
+        ":cred_helper.tar",
+        ":default_creds.tar",
+    ]
 )
 
 container_image(
     name = "structure_test_image",
     base = "@distroless_base//image",
     entrypoint = ["/structure_test"],
-    files = [":structure_test"],
+    files = [
+        ":structure_test",
+        ":cred_helper.tar",
+        ":default_creds.tar",
+    ],
 )
 
 go_prefix("github.com/GoogleCloudPlatform/container-structure-test")

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -16,6 +16,15 @@ git_repository(
     remote = "https://github.com/bazelbuild/rules_docker.git",
 )
 
+new_http_archive(
+    name = "docker_credential_gcr",
+    build_file_content = """package(default_visibility = ["//visibility:public"])
+exports_files(["docker-credential-gcr"])""",
+    sha256 = "3f02de988d69dc9c8d242b02cc10d4beb6bab151e31d63cb6af09dd604f75fce",
+    type = "tar.gz",
+    url = "https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.4.3/docker-credential-gcr_linux_amd64-1.4.3.tar.gz",
+)
+
 load(
     "@io_bazel_rules_docker//docker:docker.bzl",
     "docker_repositories",

--- a/config.json
+++ b/config.json
@@ -1,0 +1,12 @@
+{
+	"auths": {
+	},
+	"credHelpers": {
+		"appengine.gcr.io": "gcr",
+		"asia.gcr.io": "gcr",
+		"eu.gcr.io": "gcr",
+		"gcr.io": "gcr",
+		"gcr.kubernetes.io": "gcr",
+		"us.gcr.io": "gcr"
+	}
+}

